app/server.js

```diff
@@ -29,19 +29,26 @@ app.use(bodyParser.urlencoded({ extended: false }));
 app.post('/api/imgurl', function(req, res){
   //should get locale and image url
     //which will be passed into the post request
-  })
-  .then(function(err, data){
-    httpR.postAsync({
+    return httpR.postAsync({
+      headers: header,
       url: 'https://camfind.p.mashape.com/image_requests',
-      headers: header
+      'image_request[locale]': req.body.locale,
+      'image_request[remote_image_url]': req.body.imgurl
     })
-  })
   .then(function(err, data){
     //send the location and url to the server and set the data
     // data = response-data
-    return httpR.getAsync('https://camfind.p.mashape.com/image_responses/' + data)
+    return httpR.getAsync({
+      url: 'https://camfind.p.mashape.com/image_responses/' + data
+      res.end(blackBox(data.name))
+    })
+  })
+  .catch(function(e){
+    console.error('Your image is too blurry');
   })
 
+})
+
 
 var blackBox = function(string){
```

Comment:
Probably should promisify blackbox and offload the computation to an isolated worker process